includes/Product/Hooks.php

```diff
@@ -239,4 +245,61 @@ public function bulk_product_delete( $action, $products ) {
         exit;
     }
 
+    /**
+     * Triggers when admin quick edits products or bulk edit products from admin panel.
+     * we are auto selecting all category ancestors here.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @param object $product
+     *
+     * @return void
+     */
+    public function update_category_data_for_bulk_and_quick_edit( $product ) {
+        if ( ! current_user_can( 'manage_woocommerce' ) ) {
+            return;
+        }
+
+        $get_data = wp_unslash( $_REQUEST );
+
+        if ( ! isset( $get_data['woocommerce_quick_edit_nonce'] ) || ! wp_verify_nonce( sanitize_key( $get_data['woocommerce_quick_edit_nonce'] ), 'woocommerce_quick_edit_nonce' ) ) {
+            return;
+        }
+
+        $this->update_product_categories( $product->get_id() );
+    }
+
+    /**
+     * Triggers when admin saves/edits products.
+     * we are auto selecting all category ancestors here.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @param int $product_id
+     *
+     * @return void
+     */
+    public function update_category_data_for_new_and_update_product( $product_id ) {
+        if ( ! is_admin() ) {
+            return;
+        }
+
+        $this->update_product_categories( $product_id );
+    }
+
+    /**
+     * Gets chosen categories and updated product categories.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @param int $product_id
+     *
+     * @return void
+     */
+    private function update_product_categories( $product_id ) {
+        $terms             = wp_get_post_terms( $product_id, 'product_cat', [ 'fields' => 'all' ] );
```

Comment:
you only need term_id, so use fields parameter value as ids, also change Helper::generate_chosen_categories() as required